최은송/4월 4주차/BOJ_G4_1043_거짓말/BOJ_G4_1043_거짓말.java

```diff
@@ -0,0 +1,84 @@
+package boj;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+public class BOJ_G4_1043_거짓말 {
+	static List<Integer> party[], adj[];
+	static boolean[] visited, truth;
+	static int N, M, cnt = 0;
+	
+	public static void main(String[] args) throws IOException{
+		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer(in.readLine());
+		
+		N = Integer.parseInt(st.nextToken());
+		M = Integer.parseInt(st.nextToken());
+		
+		party = new LinkedList[M];
+		for(int i=0; i<M; i++) 
+			party[i] = new LinkedList<Integer>();
+		adj = new LinkedList[N+1];
+		for(int i=0; i<N+1; i++) 
+			adj[i]  = new LinkedList<Integer>();
+			
+		visited = new boolean[N+1];
+		truth = new boolean[N+1];
+		
+		st = new StringTokenizer(in.readLine());
+		int num = Integer.parseInt(st.nextToken());
+		for(int i=0; i<num; i++) {
+			truth[Integer.parseInt(st.nextToken())] = true;
+		}
+		
+		for(int i = 0; i <M; i++) {
+			st = new StringTokenizer(in.readLine());
+			num = Integer.parseInt(st.nextToken());
+			// 파티 케이스 저장 
+			for(int j=0; j<num; j++) {
+				party[i].add(Integer.parseInt(st.nextToken()));
+			}
+			// 인접한 정점들 전부 추가 
+			for(int j=0; j<num; j++) {
+				int cur = party[i].get(j);
+				for(int k=0; k<num; k++) {
+					if(j != k) {
+						adj[cur].add(party[i].get(k));
+					}
+				}
+			}
```

Comment:
이렇게 하면 중복 반복을 줄일 수 있을 것 같아요 !

```suggestion
			for(int j=0; j<num; j++) {
				int cur = Integer.parseInt(st.nextToken());
				party[i].add(cur);
				
				for(int k=0; k<j; k++) {
					adj[cur].add(party[i].get(k));
					adj[k].add(cur);
				}
			}
```

Reply:
오 그러네요! 좋은 지적 감사합니다:)